### DISCUSSION TITLE
Is there any way to use an existing viewmodel when navigating to a view?

### DISCUSSION THREAD

**Discussion by GlizzyHacker** [FilledStacks/stacked#1092] [Q&A]
I have a view which I navigate to, then I navigate back. When I navigate to it again the view creates a new viewmodel. Is there any way to have it use the existing viewmodel? I tried setting disposeViewModel to false and it works, the viewmodel isn't disposed, but the view creates a new one anyway. Thanks in advance for any answers.

**Comment by FilledStacks** [comment]
Yes, 

you can set disposeViewmodel to false, which you did, and set `createNewViewModelOnInsert` to false as well.

**Reply by GlizzyHacker** [reply]
Thanks for the reply!
I set `createNewViewModelOnInsert` to false, but I think it's false by default. I don't know if this is a bug but my issue still persists. A new ViewModel is created. 
![Képernyőkép 2024-03-26 144624](https://github.com/Stacked-Org/stacked/assets/75539035/d84da180-5639-4540-8e1f-539fa56cc54f)

I modified the default stacked project to create a demo where I am able to reproduce this.
https://github.com/GlizzyHacker/dispose_test
If I navigate back from and to `HomeView` the counter is reset. Should I create a new issue?

**Reply by FilledStacks** [reply]
Yes, that would be great. 

Please create an issue and link the repo above. We'll use that when we tackle the issue.

**Comment by GlizzyHacker** [comment]
As it seems to be a bug I created an issue
https://github.com/Stacked-Org/stacked/issues/1095
